lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,171 @@
+                       required: false,
+                       showMessages: false,
+                       titleIconName: 'fa-globe',
+                       options: this.countries,
+                       helpMessage: gettext("The country that team members primarily identify with.")
+                   });
+               },
+
+               render: function() {
+                   this.$el.html(_.template(edit_team_template)({}));
+                   this.set(this.teamNameField, '.team-required-fields');
+                   this.set(this.teamDescriptionField, '.team-required-fields');
+                   this.set(this.optionalDescriptionField, '.team-optional-fields');
+                   this.set(this.teamLanguageField, '.team-optional-fields');
+                   this.set(this.teamCountryField, '.team-optional-fields');
+                   return this;
+               },
+
+               set: function(view, selector) {
+                   var viewEl = view.$el;
+                   if (this.$(selector).has(viewEl).length) {
+                       view.render().setElement(viewEl);
+                   } else {
+                       this.$(selector).append(view.render().$el);
+                   }
+               },
+
+               createTeam: function () {
+                   var teamName = this.teamNameField.fieldValue();
+                   var teamDescription = this.teamDescriptionField.fieldValue();
+                   var teamLanguage = this.teamLanguageField.fieldValue();
+                   var teamCountry = this.teamCountryField.fieldValue();
+
+                   var data = {
+                       course_id: this.courseId,
+                       topic_id: this.topicId,
+                       name: teamName,
+                       description: teamDescription,
+                       language: _.isNull(teamLanguage) ? '' : teamLanguage,
+                       country: _.isNull(teamCountry) ? '' : teamCountry
+                   };
+
+                   var validationResult = this.validateTeamData(data);
+                   if (validationResult.status === false) {
+                       this.showMessage(validationResult.message);
+                       return;
+                   }
+
+                   var view = this;
+                   var options = {
+                       wait: true,
+                       success: function () {
+                           view.goBackToTopic();
+                       },
+                       error: function () {
+                           view.showMessage(gettext('An error occurred. Please try again.'));
+                       }
+                   };
+                   this.teamModel.save(data, options);
+               },
+
+               validateTeamData: function (data) {
+                   var status = true,
+                       message = gettext("Your team could not be created. Check the highlighted fields below and try again.");
```

Comment:
@muhammad-ammar Per our discussion, let's delete the first sentence "Your team could not be created" since it is exactly the same as the error heading.